File: src/com/gmail/marcosav2010/command/base/StopCMD.java
package com.gmail.marcosav2010.command.base;

import com.gmail.marcosav2010.command.Command;
import com.gmail.marcosav2010.main.Main;

class StopCMD extends Command {

	StopCMD() {
		super("stop", new String[] { "st", "shutdown" }, "[peer] (empty = all)");
	}

	@Override
	public void execute(String[] arg, int args) {
		if (args > 0) {
			String target = arg[0];
			Main.getInstance().getPeerManager().shutdown(target);
		} else {
			Main.getInstance().getPeerManager().shutdown();
		}
	}
}

File: cli/src/main/java/com/gmail/marcosav2010/command/base/StopCMD.kt
package com.gmail.marcosav2010.command.base

import com.gmail.marcosav2010.command.Command
import com.gmail.marcosav2010.main.Main.Companion.instance

internal class StopCMD : Command("stop", arrayOf("st", "shutdown"), "[peer] (empty = all)") {

    override fun execute(arg: Array<String>, length: Int) {
        if (length > 0) {
            val target = arg[0]
            instance.peerManager!!.shutdown(target)
        } else {
            instance.peerManager!!.shutdown()
        }
    }
}